docs/docs/ProgrammingGuide/caffe-support.md

```diff
@@ -0,0 +1,51 @@
+## **Load and Save a Caffe Model**
+
+If you have a pretrained caffe model(model definition prototxt and model binary file), you can load it into BigDL model.
+You can also convert a BigDL model to caffe model.
+
+### Load Caffe Model
+
+Assume you have a ```caffe.prototxt``` and ```caffe.model```,
+you can load it into BigDL by calling ```Module.loadCaffeModel``` (scala) or ```Model.load_caffe_model``` (python).
+
+* Scala Example
+
+```scala
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+val model = Module.loadCaffeModel(caffe.prototxt, caffe.model)
+```
+
+* Python Example
+```python
+model = Model.load_caffe_model(caffe.prototxt, caffe.model)
+```
+
+### Load Caffe Model Weights to Predefined BigDL Model
+ 
+If you have a predefined BigDL model, and want to load caffe model weights into BigDl model
+
+* Scala Example
+```scala
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+val model = Module.loadCaffe(bigdlModel, caffe.prototxt, caffe.model, matchAll = true)
+```
+Note that if ```matchAll = false```, then only layers with same name will be loaded, the rest will use initialized parameters.
+
+* Python Example
+```python
+model = Model.load_caffe_model(bigdlModel, caffe.prototxt, caffe.model, match_all=True)
+```
+
+### Save BigDL Model to Caffe Model
+* Scala Example
+```scala
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat
+bigdlModel.saveCaffe(prototxtPath, modelPath, useV2 = true, overwrite = false)
+```
+
+* Python Example
+```python
+bigdl_model.save_caffe(prototxt_path, model_path, use_v2 = True, overwrite = False)
+```
```

Comment:
we may want to light a bit about the meanings of parameters like use_v2 , overrite, etc.